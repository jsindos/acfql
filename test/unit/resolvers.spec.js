/* global expect, describe, it */

const ACFStore = require('../../generate/acfstore')
const { generateTemplate: generateResolversTemplate } = require('../../src/resolvers.template.js')

describe('generate resolvers file', () => {
  it('generateResolversTemplate', () => {
    const a = new ACFStore()
    a.addCustomPostType('wp_pear')
    a.addCustomPostType('wp_apple')
    a.addFieldGroupRelation('wp_apple', 'Apple Information')
    a.addFieldGroupRelation('wp_apple', 'Other Information')
    a.addFieldGroup('Apple Information')
    a.addField('cultivar', 'Apple Information')
    a.addField('price', 'Apple Information')
    a.addField('ripening_time', 'Apple Information')
    expect(generateResolversTemplate(a.customPostTypes, a.fieldGroups)).toMatchSnapshot()
  })
})
