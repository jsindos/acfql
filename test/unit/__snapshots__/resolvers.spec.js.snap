// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate resolvers file generateResolversTemplate 1`] = `
"function Resolvers (Connectors) {
  const Resolvers = {
    Query: {
      
      post (_, { name, id, postType }) {
        return Connectors.getPost({ name, id, postType })
      },
      posts (_, { category, postType }) {
        return Connectors.getPosts({ category, postType })
      }
    },

    Post: {
      customFields (post) {
        return Connectors.getCustomFields({ postId: post.id })
      }
    },
    
    
    
  }

  return Resolvers
}

module.exports = Resolvers
"
`;
